src/lib/device_tree/src/lib.rs

```diff
@@ -201,9 +201,10 @@ pub fn infer_type(data: &[u8]) -> Type {
     if data.len() == 0 {
         return Type::Empty;
     }
-    if let Some(i) = data.iter().position(|&c| !is_print(c)) {
-        if i == data.len() - 1 && data[i] == 0 {
-            return Type::String(unsafe { core::str::from_utf8_unchecked(&data[..data.len() - 1]) });
+    if data[..data.len() - 1] != [0] {
```

Comment:
can you clear me up on what this means? I'm confused on the [0] notation. My mind says "vector length 1 which is 0" but that's a bit different to me than " vector with 0 as value of final element" which I think is what we were thinking.
i.e., we have a vector, final element is 0, fulfilling one constraint, but we want to make sure the rest of it is value utf-8. Does this do that?

Reply:
This is just checking the following:

is element data.len() - 1 (the last element) of data equal to 0 (we use [0] to keep Rust happy).

The actual UTF8 checking has been delegated to the from_utf8() function

Reply:
Actually, we probably don't even want this check at all.

Reply:
but just checking ... isn't
data[..data.len() - 1]
a slice, as opposed to
data[data.len() - 1]

Reply:
so it matters not about the null termination? I thought Ryan wanted a NULL at the end but I no longer see why we need it ...